### ISSUE TITLE
missing error when variable in if/else is casted only in if

### ISSUE THREAD

**Issue by Horrendus** [intellij-rust/intellij-rust#5213]
## Environment

* **IntelliJ Rust plugin version:** 0.2.119.2172-193
* **Rust toolchain version:** 1.42.0 (b8cedc004 2020-03-09) x86_64-unknown-linux-gnu
* **IDE name and version:** CLion 2019.3.5 (CL-193.6911.21)
* **Operating system:** Linux 5.5.13-200.fc31.x86_64

## Problem description

The following code doesn't compile (because in sale_price in the else part, price isn't casted to an i64) but CLion with Rust plugin doesn't show an error

## Steps to reproduce
```rust
fn main() {
    let original_price = 51;
    println!("Your sale price is {}", sale_price(original_price));
}

fn sale_price(price: i32) -> i64 {
    if price > 50 {
        price as i64 - 10
    } else {
        price - 3
    }
}

```
